dev/deployment.yaml

```diff
@@ -94,7 +94,7 @@ spec:
           requiredDuringSchedulingIgnoredDuringExecution:
             nodeSelectorTerms:
               - matchExpressions:
-                  - key: bottlerocket.aws/platform-version
+                  - key: bottlerocket.aws/updater-interface-version
                     operator: Exists
```

Comment:
Quick thought (and not for this particular PR to resolve):

I think we may want to update this deployment prior to making any changes to the updater components host side. The current deployments will land on hosts that end up having a bumped version without this being scoped down to `updater-interface-version==1.0.0`. That way we can begin to vend a newer version without impacting existing deployments (which are scheduled if the key exists at all). 

I had originally aimed to have the host set its interface version by providing a compiled in default setting. If we do this then we reduce the chance that the wrong version be set for a host *and* eliminate the need for cluster operators to label their nodes.

```toml
[settings.kubernetes.node-labels]
"bottlerocket.aws/updater-interface-version" = "1.0.0"
```